siliconcompiler/core.py

```diff
@@ -4211,6 +4211,18 @@ def _local_process(self, flow, status, steplist, indexlist):
         for (step, index) in flowgraph_nodes:
             # Setting up tool is optional
             self._setup_node(step, index)
+            # Env vars are necessary as test_multiple_tools.py requires it for its version check
+            self._set_env_vars(step, index)
+            tool, task = self._get_tool_task(step, index, flow)
+            # Icarus compiles its executable during the run so we can't check its version
+            if (tool, task) != ('execute', 'exec_input'):
+                run_func = getattr(self._get_task_module(step, index, flow=flow), 'run', None)
+                try:
+                    self._check_tool_version(step, index, run_func)
+                # Convert sys.exit(1) from haltstep() in version check to SilicoCompilerError
```

Comment:
can this be changed to a `raise SiliconCompilerError` instead of `sys.exit(1)`?

Reply:
Sure I'll check if that works. That might be more elegant.

Reply:
Changed it back.
I think the reason we use `sys.exit(1)` is that this is not an error/exception in siliconcompiler so we should not show a stacktrace.